freqtrade/persistence/models.py

```diff
@@ -682,9 +684,15 @@ def update_fee(self, fee_cost: float, fee_currency: Optional[str], fee_rate: Opt
             self.fee_open_cost = fee_cost
             self.fee_open_currency = fee_currency
             if fee_rate is not None:
+                if self.fee_open_is_maker == self.fee_close_is_maker:
+                    current_rate_ratio = 1
+                elif self.fee_open_is_maker and not self.fee_close_is_maker:
+                    current_rate_ratio = 2
+                elif not self.fee_open_is_maker and self.fee_close_is_maker:
+                    current_rate_ratio = 0.5
```

Comment:
i don't think you can generalize the ratio in this way across exchanges - which makes this calculation wrong now.

It may be true for the exchange you're using (it probably isn't - if we look at binance at this moment - then fees are identical for maker/taker in the "lowest level" tier).

![image](https://user-images.githubusercontent.com/5024695/166096969-a52aacfa-f1da-43bf-858a-5586acca89f7.png)

The calculation takes the assumption that taker/maker difference is 50% - but that's only true for VIP9 users on binance (volume over 5 Billion BUSD!)...

Reply:
Right, we should pass in the taker and maker ratio instead.

It's true for binance futures, but yeah I was generalizing too much